keps/sig-autoscaling/3148-hpa-dryrun/README.md

```diff
@@ -0,0 +1,890 @@
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+Currently the HPA does not support a dry-run mode. 
+With scaling decisions based on built-in/custom metrics there are use cases when one wants to run HPA without actually changing the number of pods running (to test the scaling decision, validate/approve it, etc). The proposal focuses on enabling/disabling a HPA dry-run mode using a `updatePolicy.updateMode` field. It is a low risk/low cost change that can allow users to use HPA to benefit their custom scaling scenarios.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+To be able to get the insights on pod scaling mechanisms, one needs to be able to run HPA without actually rescaling pods. In cases when scaling mechanisms (based on in-built/custom metrics) include multiple stages such as testing the scaling decision, validating and approving it (whether the output of scaling mechanism is correct) - being able to dry-run HPA seems benefiacial.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- allow users to run dry-run HPA through setting updateMode off
+- allow users to set dry-run mode on and off on a running cluster
+- allow users to see the inforamtion on proposed scaling solution
+
+### Non-Goals
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+TBD
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+There should be a way to enable dry-run mode when creating autoscaler, such as:
+
+
+```
+kubectl autoscale deployment php-apache --cpu-percent=50 --updateMode=off
+```
+
+The default `updateMode` should be set to `Auto`, so the user can set it off if it is needed.
+
+Users should be able to change `updatePolicy.updateMode` with running HPA through editing HPA resource.
+
+A typical example `yaml` file to describe HPA looks like following 
+
+```yaml
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+spec:
+  scaleTargetRef:
+	apiVersion: apps/v1
```

Comment:
These tabs render as dark red stripes.  Perhaps you should change them back to spaces?  Take a look at the rendered markdown to make sure it looks okay.

![image](https://user-images.githubusercontent.com/1103629/151982138-aa55a6fb-2ab6-433a-bc39-3feb0b16c503.png)